port-chooser/src/main/java/org/terracotta/utilities/test/net/PortManager.java

```diff
@@ -159,6 +164,55 @@ private PortManager() {
     }
   }
 
+  /**
+   * Indicates if the designated port is in the range of ports that <i>may</i> be allocated
+   * by this class.  If the designated port is in the range of ports that may be allocated,
+   * the {@link #reserve(int)} method may be used to attempt to allocate the port.  The port
+   * may also be returned from the {@link #reservePort()} and {@link #reservePorts(int)} methods.
+   * @throws IllegalArgumentException if {@code port} is not between 0 and
+   *    {@value #MAXIMUM_PORT_NUMBER} (inclusive)
+   */
+  public boolean isReservablePort(int port) {
+    if (port < 0 || port > MAXIMUM_PORT_NUMBER) {
+      throw new IllegalArgumentException("Port " + port + " is not a valid port number");
+    }
+    return !restrictedPorts.get(port);
+  }
+
+  /**
+   * Gets the <i>active</i> {@link PortRef} instance for the designated port.
+   * <p>
+   * This method returns a reference to the most recent {@code PortRef} created
+   * for the designated port if the {@code PortRef} is both <i>strongly reachable</i>
+   * and not closed.  The result of this method may be immediately stale -- the
+   * {@code PortRef} may be closed between the time the {@code PortRef} is checked
+   * and the reference returned to the called.
+   * @param port the port number for which the {@code PortRef} is returned
+   * @return an {@code Optional} of the {@code PortRef} for {@code port} if the
+   *      {@code PortRef} is both strongly reachable and not closed;
+   *      {@code Optional.empty} if the {@code PortRef} for {@code port} is either
+   *      not strongly reachable or closed
+   * @throws IllegalArgumentException if {@code port} is not between 0 and
+   *    {@value #MAXIMUM_PORT_NUMBER} (inclusive) or is not a reservable port
+   */
+  public synchronized Optional<PortRef> getPortRef(int port) {
+    if (port < 0 || port > MAXIMUM_PORT_NUMBER) {
+      throw new IllegalArgumentException("Port " + port + " is not a valid port number");
+    }
+    if (restrictedPorts.get(port)) {
+      throw new IllegalArgumentException("Port " + port + " is not reservable");
+    }
+
+    cleanReleasedPorts();
```

Comment:
I think you can drop these 4 lines.

Reply:
I agree we can get rid of the `if` but I'm pretty sure the call to `cleanReleasedPorts` needs to be retained.